dali/pipeline/executor/executor.h

```diff
@@ -118,6 +138,60 @@ class DLL_PUBLIC Executor : public ExecutorBase, public WorkspacePolicy, public
   DISABLE_COPY_MOVE_ASSIGN(Executor);
 
  protected:
+  template<typename backend>
+  inline void GetMaxSizes(TensorList<backend> &in, size_t &max_out_size,
+                          size_t &max_reserved_size) {
+    auto out_size = in.nbytes();
+    auto reserved_size = in.capacity();
+    max_out_size = std::max<size_t>(std::ceil((out_size * 1.0) / in.ntensor()), max_out_size);
+    max_reserved_size = std::max<size_t>(std::ceil((reserved_size * 1.0) / in.ntensor()),
+                                         max_reserved_size);
+  }
+  template<typename backend>
+  inline void GetMaxSizes(TensorVector<backend> &in, size_t &max_out_size,
+                          size_t &max_reserved_size) {
+    for (size_t j = 0; j < in.ntensor(); ++j) {
+      max_out_size = std::max(in[j].nbytes(), max_out_size);
+      max_reserved_size = std::max(in[j].capacity(), max_reserved_size);
```

Comment:
I have some reservations regarding this. In the case of contiguous TensorVector, the Tensors that you're accessing with operator[] here are kind of views into the backing TensorList. So the `capacity` will probably match exactly the `nbytes` here, but the total capacity of the TensorVector _might_ be bigger than `nsamples * max(number of bytes)` (I suspect). 

This can be probably checked with a test that sets the TensorVector to contiguous mode, resizes it to some big shapes and than sets all the shapes to be for example half the initial size.

Can you check this? I think the best solution would be to take max of this reserved size and the average reserved size you calculate for the TensorList in function above.